### PR TITLE
[Enhance] Rename the function `upsample_like` to `interpolate_as`

### PR DESCRIPTION
**Pull request by AronLin**
Actually, this function can not only upsample the input Tensor but also downsample the input.
So we should rename it to reflect its real function.

### CODE REVIEW

File: mmdet/models/utils/misc.py [hunk 1]
```diff
@@ -1,29 +1,29 @@
 from torch.nn import functional as F
 
 
-def upsample_like(source, target, mode='bilinear', align_corners=False):
-    """Upsample the source to the shape of the target.
+def interpolate_as(source, target, mode='bilinear', align_corners=False):
+    """Down/up samples the source to the shape of the target.
```
Comment (ZwwWayne) [inline mmdet/models/utils/misc.py:5]: Interpolate the source

**Review by ZwwWayne — APPROVED** [review]

File: mmdet/models/utils/misc.py [hunk 2]
```diff
@@ -1,29 +1,29 @@
 from torch.nn import functional as F
 
 
-def upsample_like(source, target, mode='bilinear', align_corners=False):
-    """Upsample the source to the shape of the target.
+def interpolate_as(source, target, mode='bilinear', align_corners=False):
```
Comment (jshilong) [inline mmdet/models/utils/misc.py:4]:
```
Interpolate the `source` to the shape of the `target`.

The `source` must be a Tensor, but the `target` can be a Tensor or a np.ndarray with the shape
(..., target_h, target_w).
```

**Review by jshilong — APPROVED** [review]
Please resolve my comments first

File: mmdet/models/utils/misc.py (hunk 2 above)
Comment (AronLin) [reply mmdet/models/utils/misc.py:4]: ok, I will resolve it in #5577.

**Pull request merged**